src/Core/Models/Domain/LoginUri.cs

```diff
@@ -35,5 +44,21 @@ public LoginUriData ToLoginUriData()
             BuildDataModel(this, u, _map, new HashSet<string> { "Match" });
             return u;
         }
+
+        public async Task<bool> ValidateChecksum(string clearTextUri, string orgId, SymmetricCryptoKey key)
+        {
+            if (this.UriChecksum == null)
+            {
+                return false;
+            }
+
+            // HACK: I don't like resolving this here but I can't see a better way without
+            // refactoring a lot of things.
+            var cryptoService = ServiceContainer.Resolve<ICryptoService>();
+            var localChecksum = await cryptoService.HashAsync(clearTextUri, CryptoHashAlgorithm.Sha256);
+
+            var remoteChecksum = await this.UriChecksum.DecryptAsync(orgId, key);
+            return remoteChecksum == localChecksum;
+        }
```

Comment:
⛏️ Remove the use of `this` to be consistent with the app code.

❓ In case an exception happens, should this break the whole decryption or just log and return `false`?

🤔 I don't like resolving that there neither. I was thinking about resolving it in one the parents and then pass it along here, so if the user has many Uris it doesn't have to resolve it many times; but it's also dirty. Also thought about moving it to a field and make it `LazyResolve` but I think this happens only once per instance so maybe it doesn't help much. The only thing that comes to mind to improve a bit the performance would be to move this check to the `CryptoService` and then resolve such service in the `Login` parent and call it from there in the for, so the service is resolved just once for all Uris of one login, furthermore we don't pollute even more the models.
```c#
// In `ICryptoService`
Task<bool> ValidateUriChecksumAsync(EncString remoteUriChecksum, string rawUri, string orgId, SymmetricCryptoKey key)
```
Then in `Login`
```c#
if (Uris != null)
{
    var cryptoService = ServiceContainer.Resolve<ICryptoService>();
    view.Uris = new List<LoginUriView>();
    foreach (var uri in Uris)
    {
        view.Uris.Add(await uri.DecryptAsync(orgId, key));
        var loginUriView = await uri.DecryptAsync(orgId, key);
        if (bypassValidation || await cryptoService.ValidateUriChecksumAsync(uri.UriChecksum, loginUriView.Uri, orgId, key))
            {
            view.Uris.Add(loginUriView);
        }
    }
}
```